docs/reference/setup/add-nodes.asciidoc

```diff
@@ -41,3 +41,141 @@ the rest of its cluster.
 
 For more information about discovery and shard allocation, see
 <<modules-discovery>> and <<modules-cluster>>.
+
```

Comment:
Note to reviewers: This content is from https://github.com/elastic/elasticsearch/pull/56611/files#diff-e8465aac40349b409bae92fb299814f3 with no substantive changes.

Only `[discrete]` tags were added.